### DISCUSSION TITLE
Welcome to autoortho Discussions!

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
 
 * Please keep the conversations civil and remember that everyone here is volunteering their time
 * If you have an issue, please look for existing solutions first in dicussions or 'Issues' above
 * If you have a new issue, please make a brand new issue.  (See the 'Issues' tab above) rather than comment on an unrelated issue or thread.

**Comment by ZainA-Official** [comment]
Auto ortho shows as a trojan virus on my PC, is it or is that a bug?

**Comment by kubilus1** [comment]
That's just Windows for you.

**Reply by ZainA-Official** [reply]
Ok, I will download it.

**Comment by ILANKRT** [comment]
I am very disappointed about the missing autoortho for Mac (I own M! Mac Studio). I understand that you may not have a Mac on your desk but it is important to let us take part on your work. I am not a programmer so I cannot give advices for compilation, maybe you know someone else .. Thanks

**Reply by kubilus1** [reply]
I did pickup a mac book not too long ago but haven't had much time to work on this for the moment.  Eventually though, I'd like to make a mac release as well.

**Comment by ILANKRT** [comment]
Thanks for quick answer. I hope to use your add-on in the near future…


Ilan Dr. Kreitmann
***@***.***

**Comment by Peppone220** [comment]
Congratulations on the work done with AutoOrtho, simply brilliant.
I installed everything on my computer no Special problems, but the cities are not in 3D but flat:  how can I solve?
I attach Images and the Scenery_packs.ini file, please help me.
[scenery_packs.ini.txt](https://github.com/kubilus1/autoortho/files/12063644/scenery_packs.ini.txt)
![5 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/b89146a3-e4f5-4519-a951-ef25a0fd2764)
![4 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/a88f1403-bd2f-4125-b4aa-c8c851f223df)
![3 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/48c9b2c7-7fc3-4749-958a-a51d2f49fce3)
![2 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/6e0e6135-64ff-422b-bb36-7ea7d9e721db)
![1 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/7ff1809b-c9f2-4da2-841f-cfef725db6e7)

**Reply by dpdurst** [reply]
Hint, start with this entry and disable everything below it and see if scenery then shows up. If it does then you should be able to figure out which entry is causing your issue.

SCENERY_PACK_DISABLED Custom Scenery/prague/

**Comment by kubilus1** [comment]
I see some issues in that scenery_packs file, you may need to cleanup a
bit.  For one 'SCENERY_PACK Custom Scenery/yOrtho4XP_Overlays/' is listed
at the bottom which seems wrong as well.

**Reply by Peppone220** [reply]
OK, many thanks for the quick response, I'll try to follow your advice and as soon as it's resolved, I'll let you know

**Reply by Peppone220** [reply]
I am desperate, I followed your advice but nothing happens and I don't understand why the scenario is absolutely unrealistic and always the same. I also deleted all the Lbrary from the Custom Scenery folder but, even thus AutoOrtho does not work.
I send the contents of the custom scenery folder to make you understand what's inside. I also send the latest Scenery_packs.ini file to be able to check it.
Now I also try to eliminate AutoOrtho
 and install it again so I also try this solution.
I need to be helped
[scenery_packs.ini2.txt](https://github.com/kubilus1/autoortho/files/12064571/scenery_packs.ini2.txt)
![1 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/0178d7d2-f850-485a-80c3-36f908b30283)
![6 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/70e57493-5dc9-46e2-8e26-0ac123816f93)
![7 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/07dde8a7-2495-406b-9c6d-6064e856635e)
![10 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/23b8f5af-0eba-4062-8b4e-09c243d1e672)
![11 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/c03cd737-9dad-417c-a8e9-8ecf3e5fa04a)
![12 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/74d15c03-0693-47ff-a5f6-058fefb0f4d9)
![13 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/6f3f1c69-804b-484c-a332-e261b8aab9ed)
![14 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/ce3d1b46-4f77-4a8e-b50d-d3aa643736b6)
![15 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/9df657c9-85df-4410-81e5-f679ff37928b)
![16  Cattura](https://github.com/kubilus1/autoortho/assets/74368213/5f244bdc-9f4d-4653-96c1-8a0275a2d362)
![16 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/553fb162-d11d-40d3-84fd-0130e60a7203)
![17 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/a6c1dd66-9ead-4342-998f-e6531b23a06c)

**Reply by Peppone220** [reply]
Good day and thanks for following my problemathe, unfortunately I was unable to solve the problems with Autotoho.
As you can see from the images and attached files I also made the attempt to remove all the folders of the Custom Scenery airports, but nothing changes. I always have the same scenario as my city in Portugal that does not resemble reality.
I have the feeling that the tiles are not loaded and I don't understand why.
I don't know how to come out of this problem.
Sorry if I send you a lot of material but it is to give you the opportunity to study what's wrong on my computer.
Thanks in advance for your assistance, and good day
[scenery_packs.ini- 17-07-2023.txt](https://github.com/kubilus1/autoortho/files/12072369/scenery_packs.ini-.17-07-2023.txt)
![20 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/1f7e6363-5f02-4136-9125-a54f21e5b756)
![21 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/d1c0baf4-b14d-4244-b4cc-ef626f002c02)
![22 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/f7a84bb9-9904-46d0-a6c2-e6d7fdb07a4e)
![23 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/0fe44a4e-df91-450c-911e-689ecfc4162f)
![24 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/3d01b0a9-defa-4479-b7e8-d15152fc4db2)
![26 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/3888c9a0-5332-41f5-9e0f-16bdb77f8999)
![27 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/12d35d14-a761-4839-b95b-3b2a8f0935f2)
![28 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/f315c605-8f24-48a9-88bd-485b766494f2)
![29 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/a6532b76-c345-4665-abf4-364167bc8950)
![30 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/3b720ad1-35d2-410f-a9c7-7c719db955ee)
![31 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/eea45e0a-f233-484e-be11-744d94970ae5)
![33 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/76f06012-5237-4824-aa00-3865fb735191)
![36 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/7121e8bc-9be1-4d1f-8daa-63c8bbc0482b)
![38 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/6d20855e-e4a2-4a1f-96f5-acaf79e84f1b)
![39 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/789f7de8-43f1-4584-aeb8-58617778bc13)
![40 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/986bc3e8-e0b2-430f-b242-eaf985df51f0)
![41 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/6774ad5e-703d-4df9-8029-f86f16a41754)

**Comment by dpdurst** [comment]
Try this you have to have the AutoOrtho stuff below the Global Scenery entry in the .ini file. This should show now the houses etc.

Remove the .txt at the end and put in your Custom Scenery folder and test

[scenery_packs.ini.txt](https://github.com/kubilus1/autoortho/files/12065204/scenery_packs.ini.txt)

**Comment by dpdurst** [comment]
What exactly are you expecting? From what I see  auto ortho is working, If you expecting a 100% identical look down to the houses you should rethink those expectations. Maybe AO is not for you if you don't like the way things look.

**Reply by Peppone220** [reply]
Ok I understand, we can't do better, I have reinstalled all the other libraries and now the appearance is a little more realistic, I think the limit is the Bing Tiles, it would be better to be able to work with those of Google, which however are not currently usable.
Thanks anyway for the help

**Comment by kubilus1** [comment]
I'm a little confused about the expectations.  It looks like this project is working just like it should.  If you want better overlays (the houses etc) you may want to checkout SimHeaven:  https://simheaven.com/simdownloads/x-world-europe/

**Reply by Peppone220** [reply]
Ok I understand, we can't do better, I have reinstalled all the other libraries and now the appearance is a little more realistic, I think the limit is the Bing Tiles, it would be better to be able to work with those of Google, which however are not currently usable.
Thanks anyway for the help
![51 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/e1492a4a-ec44-46ba-a0f1-342a87214c72)
![52 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/ec08e975-f827-4401-b5a0-1819417561ba)
![53 Cattura](https://github.com/kubilus1/autoortho/assets/74368213/5e5c492e-97fc-40ba-adc6-abf3b848f763)
![54 1 00 frrt Cattura](https://github.com/kubilus1/autoortho/assets/74368213/4fe00cc2-b49f-4a3f-b2e6-6852de27b87b)

**Comment by Ertzwert** [comment]
Having fun for several days with autoortho ( some times better then the MSFS "thing" )
 but since yesterday if I start autoortho the cmd window pops up for a fraction of a second  and thats unfortunatly it.
Not working anymore .. and also no Idea what happend or how to solve this.
I removed ( also in regedit ) I reinstalled ... but it is gone  :-(  
appreciate your help
Thnkx in advance

**Comment by kubilus1** [comment]
Try running by first opening a command prompt or PowerShell first.  Then
you may be able to see what the error is.

**Reply by Ertzwert** [reply]
Thankx .. I got this   but no clue why  
is it a phyton thing ? in the os.py ? 
but I cant find the F:\AUTOOR~2\AUTOOR~2\os.py", 
something with my path .. how to correct ?

Config file found C:\Users\xxx \.autoortho reading...
Traceback (most recent call last):
  File "F:\AUTOOR~2\AUTOOR~2\__main__.py", line 6, in <module>
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "F:\AUTOOR~2\AUTOOR~2\aoconfig.py", line 491, in <module aoconfig>
  File "F:\AUTOOR~2\AUTOOR~2\aoconfig.py", line 436, in __init__
  File "F:\AUTOOR~2\AUTOOR~2\aoconfig.py", line 451, in load
  File "F:\AUTOOR~2\AUTOOR~2\aoconfig.py", line 468, in get_config
  File "F:\AUTOOR~2\AUTOOR~2\os.py", line 215, in makedirs
  File "F:\AUTOOR~2\AUTOOR~2\os.py", line 215, in makedirs
  File "F:\AUTOOR~2\AUTOOR~2\os.py", line 215, in makedirs
  File "F:\AUTOOR~2\AUTOOR~2\os.py", line 225, in makedirs
OSError: [WinError 123] Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch: 'xxxxxxxxx:'

**Reply by Ertzwert** [reply]
Something here .. ( I am not an python expert )

def makedirs(name, mode=0o777, exist_ok=False):
    """makedirs(name [, mode=0o777][, exist_ok=False])

    Super-mkdir; create a leaf directory and all intermediate ones.  Works like
    mkdir, except that any intermediate path segment (not just the rightmost)
    will be created if it does not exist. If the target directory already
    exists, raise an OSError if exist_ok is False. Otherwise no exception is
    raised.  This is recursive.

    """
    head, tail = path.split(name)
    if not tail:
        head, tail = path.split(head)
    if head and tail and not path.exists(head):
        try:
            makedirs(head, exist_ok=exist_ok)
        except FileExistsError:

**Comment by Ertzwert** [comment]
All solved !!! 2 day later found the culprit 

The problem was caused by me in the .autoortho config file 

I think I opened it and tipped instead of somewhere else in the  x-plane custom scenery path a lot of x'es in the wrong place

scenery_path = xxxxxxxxxF:/X-Plane 12/Custom Scenery

so no wonder 
sorry for the confusion

**Reply by dpdurst** [reply]
LOL we've all done stuff like this, glad you found it and its all resolved!

**Comment by Ertzwert** [comment]
oh yes sometimes it feels like :
 " oh my bike is not working and you start taking the engine apart / contact (confuse) the manufacturer . finding nothing  and put it together until someone asks you if you've actually filled the tank .....  :-)

**Comment by Ertzwert** [comment]
Now I stepped over to 0,5.  but having issuses  allthouhg after clearing my prvious problem ... everthing looked normal 
Installation ok / scenery download ok etc 
I can run autoortho ...  some tiles are loaded but then X-plane 12 is in "endless loop " (see log ) 
(on PC restart for exiting XP ) 
Maybe again som path problem =  attached the log and config 

Thanky for help

[autoortho.log](https://github.com/kubilus1/autoortho/files/12139181/autoortho.log)
[autoortho.txt](https://github.com/kubilus1/autoortho/files/12139198/autoortho.txt)

**Comment by gssev** [comment]
Thanks for AutoOrtho, Massive thank you to the author for autoortho. I have it working with XP12 and has made xplane much more better and enjoyable. Xplane should have this integrated in it.  Is there a donation page?
Regards
Graeme
